### PR TITLE
Improve JWT validator configurations support

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
This PR improves the existing JWT validation configuration support further.

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/1240

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: jwt-ballerina/jwt_validator.bal [hunk 1]
```diff
@@ -523,12 +558,47 @@ isolated function validateAudience(Payload payload, string|string[] audienceConf
...
+        return prepareError("JWT must contain a valid JWT ID.");
+    }
+}
+
+isolated function validateKeyId(Header header, string keyIdConfig) returns Error? {
+    string? keyIdHeader = header?.kid;
+    if (keyIdHeader is string) {
+        if (keyIdHeader != keyIdConfig) {
+            return prepareError("JWT contained invalid key ID '" + keyIdHeader + "'");
+        }
```
Comment (ayeshLK) [inline jwt-ballerina/jwt_validator.bal:583]:
```suggestion
    if (keyIdHeader is string && keyIdHeader != keyIdConfig) {
        return prepareError("JWT contained invalid key ID '" + keyIdHeader + "'");
```

File: jwt-ballerina/jwt_validator.bal [hunk 2]
```diff
@@ -523,12 +558,47 @@ isolated function validateAudience(Payload payload, string|string[] audienceConf
...
         return prepareError("JWT must contain a valid audience.");
     }
 }
 
+isolated function validateJwtId(Payload payload, string jwtIdConfig) returns Error? {
+    string? jwtIdPayload = payload?.jti;
+    if (jwtIdPayload is string) {
+        if (jwtIdPayload != jwtIdConfig) {
+            return prepareError("JWT contained invalid JWT ID '" + jwtIdPayload + "'");
+        }
```
Comment (ayeshLK) [inline jwt-ballerina/jwt_validator.bal:572]:
```suggestion
    if (jwtIdPayload is string && jwtIdPayload != jwtIdConfig) {
        return prepareError("JWT contained invalid JWT ID '" + jwtIdPayload + "'");
```

File: jwt-ballerina/jwt_validator.bal [hunk 3]
```diff
@@ -481,6 +505,17 @@ isolated function assertSignature(SigningAlgorithm alg, byte[] assertion, byte[]
     return prepareError("Unsupported JWS algorithm '" + alg.toString() + "'.");
 }
 
+isolated function validateUsername(Payload payload, string usernameConfig) returns Error? {
+    string? usernamePayload = payload?.sub;
+    if (usernamePayload is string) {
+        if (usernamePayload != usernameConfig) {
+            return prepareError("JWT contained invalid username '" + usernamePayload + "'");
+        }
```
Comment (ayeshLK) [inline jwt-ballerina/jwt_validator.bal:513]:
```suggestion
    if (usernamePayload is string && usernamePayload != usernameConfig) {
        return prepareError("JWT contained invalid username '" + usernamePayload + "'");
```

**Review by ayeshLK — CHANGES_REQUESTED** [review]

File: jwt-ballerina/jwt_validator.bal (hunk 1 above)
Comment (ldclakmal) [reply jwt-ballerina/jwt_validator.bal:583]: @ayeshLK Here I want to simply ignore the following case.
```ballerina
keyIdHeader is string and keyIdHeader == keyIdConfig
```
With the suggested approach, the above case will definitely go to the else part. But there I want to handle `keyIdHeader is not string` separately. So, there need to be another `if` condition. WDYT?

File: jwt-ballerina/jwt_validator.bal (hunk 2 above)
Comment (ldclakmal) [reply jwt-ballerina/jwt_validator.bal:572]: Same as the above comment.

File: jwt-ballerina/jwt_validator.bal (hunk 3 above)
Comment (ldclakmal) [reply jwt-ballerina/jwt_validator.bal:513]: Same as the above comment.

**Review by ayeshLK — APPROVED** [review]

**Pull request merged**